rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport repo](https://github.com/gravitational/teleport)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport/labels/ui) to
+        those transferred issues.
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PRs.
+  - [ ] Apply the necessary backports.
+
+### Webapps.e repository
+
+[repository](https://github.com/gravitational/webapps.e)
+
+#### Actions
+
+- [ ] Triage issues.
+  - [ ] Close those that are no longer relevant.
+  - [ ] [Transfer remaining issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository) to the [Teleport.e repo](https://github.com/gravitational/teleport.e)
+  - [ ] Add the [ui label](https://github.com/gravitational/teleport.e/labels/ui)
+- [ ] Triage PRs.
+  - [ ] Close those that are no longer relevant.
+  - [ ] Merge remaining PR's.
+  - [ ] Apply the necessary backports.
+
+### Teleport repository
+
+[repository](https://github.com/gravitational/teleport)
+
+#### Actions
+
+- Remove `/webassets` submodule
+  - This submodule is no longer required as the web UI will be built on demand.
+- Clone the [Webapps repository](https://github.com/gravitational/webapps) into
+  the Teleport root. [Maintaining their respective git histories](https://stackoverflow.com/questions/13040958/merge-two-git-repositories-without-breaking-file-history)
+  - This will need to be done for every respective version branch (v9, v10, v11)
+- update targets that use the `packages/webapps.e` submodule to point points to
+  the correct version in the `teleport.e/webapps` folder.
+- only require teleport build processes to run on teleport paths and the webapp
+  ones to run on the webapp paths
+-
+- Archive the [Webapps repository](https://github.com/gravitational/webapps).
+
+#### CI jobs
+
+- **CodeQL / Analyze (go) (pull_request)**
+- **Teleport-IntegrationTest (ci-account)**
+- **Teleport-UnitTest (ci-account)**
+- **Teleport-DocTest (ci-account)**
+- **Teleport-Lint (ci-account)**
+- **Teleport-OsCompatibility (ci-account)**
+  - Disable for changes exclusively made in the `/webapp` path.
+- **Assign / Auto Request Review**
+  - Create a list of full stack engineers and have it pick from this list for
+    changes made to the `/webapp` path.
+- **Check / Checking reviewers**
+  - Continue to require 2 reviewers for changes in the `webapp` projects.
+- **Label / Label Pull Request (pull_request_target)**
+  - Add label for UI to changes made in the `/webapp` path.
+- **CodeQL / Analyze (javascript) (pull_request)**
+  - Ensure that it's running for changes in `/webapp` path.
+- **Code scanning results / CodeQL**
+  - No changes
+- **webapps-build**
+  - Instead of having this job build webassets it should ensure that it can be
+    built.
```

Comment:
Isn't ensuring that webassets can be built the thing that this job does already?

Also, the `push-build-darwin-amd64` pipeline already runs `yarn install`, `yarn build-term` and `yarn package-term`. However, this only happens after a commit lands in teleport's master. I don't know if we do it for other platforms as well, we'd need to look into the dronegen stuff. The question is whether we want the push pipelines to still run on every commit to teleport.

The reason Connect is build during `push-build-darwin-amd64` is that it's one of the artifacts published on tag release. If there are problems with building Connect, [we want to discover them ASAP rather than wait for a tag release to break](https://gravitational.slack.com/archives/C0DF0TPMY/p1654618901893779?thread_ts=1654598627.161199&cid=C0DF0TPMY). Though TBH, at the moment the Connect build being done on push is somewhat different from the build being done on a tag release, but that's another story.

Reply:
What I meant to say was that it can just try to build the webassets not build and push. I'll update the copy.

Reply:
Are we talking about the webapps-build CI job from `.cloudbuild/ci/build.yaml`? I don't think it pushes anything right now. 🤔

Reply:
Ping; I don't think this jobs pushes webassets?

Reply:
It's still not clear to me what kind of pushing this job does and will not do in the future. Am I mistaking it with some other job?